lib/ansible/modules/source_control/git.py

```diff
@@ -1023,13 +1058,25 @@ def main():
     # call run_command()
     module.run_command_environ_update = dict(LANG='C', LC_ALL='C', LC_MESSAGES='C', LC_CTYPE='C')
 
+    if separate_git_dir is not None and bare:
```

Comment:
We probably want to fail loudly and there's [`mutually_exclusive=`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/packaging/os/portage.py#L492-L496) argument for this case. Please use it.